### DISCUSSION TITLE
Why doesn't ScopesAuthorizer find my scope?

### DISCUSSION THREAD

**Discussion by bednart** [ThreeMammals/Ocelot#1878] [Q&A] [answered]
Hi,
I'm trying to use AllowedScopes array in Ocelot config with only one value: 
`"AllowedScopes": [ "access_as_user" ]`

I'm using Azure AD App Registration with specified scope at "Expose an API" tab and there is **scp** claim in JWT access token once requested: 
`"scp": "access_as_user"`

Problem is that [ScopeAuthorizer](https://github.com/ThreeMammals/Ocelot/blob/36ad6e1e7873526e28300c7ac6312052f6df14e9/src/Ocelot/Authorization/ScopesAuthorizer.cs#L27) uses build in "**scope**" as claim type, but access_as_user is in claim with "**http://schemas.microsoft.com/identity/claims/scope**" type so it is not found and ScopesAuthorizer throws error "no one user scope: '' match with some allowed scope: 'access_as_user'"

Question is. Am I doing something wrong or AllowedScopes array is not usable for this use case?

Thanks,
beda

**Comment by vitorelli** [comment]
Hello @bednart ,

I faced a similar issue. Please check if you are using the Authentication midleware `app.UseAuthentication()` or\and `JwtSecurityTokenHandler.DefaultInboundClaimTypeMap` to map your scopes.

You may find more information here: https://ocelot.readthedocs.io/en/latest/features/authentication.html

**Reply by raman-m** [reply]
+ References from #1876

**Comment by raman-m** [comment, marked as answer]
> @bednart wrote [on Mar 7, 2022](https://github.com/ThreeMammals/Ocelot/discussions/1878#discussion-6034588)

Hi Tomáš!
In addition to the message above from @vitorelli, please look into this issue:
- #1876
